Thermal electronic contribution

nhew1994 opened the pull request:
For Al:
![Fel](https://github.com/user-attachments/assets/c2bd56be-9b9a-4bc3-a657-457e07a87c66)
![Sel](https://github.com/user-attachments/assets/a64a2529-dd64-4634-a6a9-4589dd6e9d1f)
![Cv,el](https://github.com/user-attachments/assets/d0af89a5-d1bf-4cd6-9047-b639789fa17e)
![Fel_fit](https://github.com/user-attachments/assets/99a01e12-154e-4fa0-aa08-8429376162c9)
![Sel_fit](https://github.com/user-attachments/assets/5fc590d4-e396-453d-8d12-7994f517449f)
![Cv,el_fit](https://github.com/user-attachments/assets/a881c7da-d442-4029-ad0b-128d73026634)

I could use some help deriving the equation for Cv. Dr. Shang said he just used Cv = T dS/dT.

nhew1994 committed:
Added new thermal_electronic.py file

nhew1994 committed:
Update workflows.py and thermal_electronic.py

workflows.py
- Increased default NEDOS from 5001 to 10001
- Added TODOs

thermal_electronic.py
- renamed k_B to BOLTZMANN_CONSTANT
- added TODOs

lukeamyers reviewed (approved):
I made some suggestions. please merge whenever you are satisfied.

nhew1994 committed:
Added a TODO to implement for magnetic materials

nhew1994 committed:
Decoupled plotting from read_total_electron_dos into plot_total_electron_dos and from fermi_dirac_distribution into plot_fermi_dirac_distribution

nhew1994 committed:
Added min_chemical_potential and max_chemical_pote
ntial as optional arguments to def calculate_chemical_potential

nhew1994 committed:
Implemented np.where in calculate_entropy and calculate_heat_capacity.
Also added plot_entropy_integral and plot_heat_capacity_integral to
for visualization.

nhew1994 committed:
In workflows.py, made NEDOS an optional parameter
with default 10001 in def run_elec_dos

nhew1994 committed:
Added docstrings to thermal_electronic.py
